### DISCUSSION TITLE
Mutual TLS

### DISCUSSION THREAD

**Discussion by cgb4** [uNetworking/uWebSockets#1408] [General]
Thank you for this excellent project.

For Mutual TLS we would like to optionally set SSL_VERIFY_FAIL_IF_NO_PEER_CERT in the call to SSL_CTX_set_verify in openssl.c like this:

SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER | SSL_VERIFY_IF_NO_PEER_CERT, NULL);

Possibly a boolean _ssl_verify_client_ flag could be added to SocketContextOptions to allow this.

**Comment by ghost** [comment]
There is getNativeContext which returns SSL_CTX you can do anything with
